caluma/caluma_form/tests/test_answer.py

```diff
@@ -333,3 +334,70 @@ def validate(self, mutation, data, info):
         result.data["saveDocumentStringAnswer"]["answer"]["stringValue"]
         == "Test (validated)"
     )
+
+
+@pytest.mark.parametrize("question__type", ["file"])
+def test_file_answer_metadata(db, answer, schema_executor, minio_mock):
+    query = """
+        query ans($id: ID!) {
+            node(id:$id) {
+                ... on FileAnswer {
+                    fileValue: value {
+                        name
+                        downloadUrl
+                        metadata
+                    }
+                }
+            }
+        }
+    """
+    vars = {"id": to_global_id("FileAnswer", str(answer.pk))}
+
+    # before "upload"
+    old_stat = minio_mock.stat_object.return_value
+    minio_mock.stat_object.side_effect = minio.error.S3Error(
+        "NoSuchKey",
+        "object does not exist",
+        resource="bla",
+        request_id=134,
+        host_id="minio",
+        response="bla",
+    )
+
+    # Before upload, no metadata is available
+    result = schema_executor(query, variable_values=vars)
+    assert not result.errors
+    assert result.data["node"]["fileValue"]["metadata"] is None
+
+    # After "upload", metadata should contain some useful data
+    minio_mock.stat_object.return_value = old_stat
+    minio_mock.stat_object.side_effect = None
+
+    result = schema_executor(query, variable_values=vars)
+    assert not result.errors
+
+    # Make sure all the values (especially in metadata) are JSON serializable.
+    # This is something the schema_executor doesn't test, but may bite us in prod
+    metadata = result.data["node"]["fileValue"]["metadata"]
+
+    assert metadata == {
+        "bucket_name": "caluma-media",
+        "content_type": "application/pdf",
+        "etag": "5d41402abc4b2a76b9719d911017c592",
+        "last_modified": "2021-03-05T15:24:33+00:00",
+        "metadata": {
+            "Accept-Ranges": "bytes",
+            "Content-Length": "5",
+            "Content-Security-Policy": "block-all-mixed-content",
+            "Content-Type": "binary/octet-stream",
+            "Date": "Fri, 05 Mar 2021 15:25:15 GMT",
+            "ETag": '"5d41402abc4b2a76b9719d911017c592"',
+            "Last-Modified": "Fri, 05 Mar 2021 15:24:33 GMT",
+            "Server": "MinIO",
+            "X-Amz-Request-Id": "16697BAAD69D2214",
+            "X-Xss-Protection": "1; mode=block",
+            "Vary": "Origin",
+            "Content-Type": "binary/octet-stream",
+        },
+        "size": 8200,
+    }
```

Comment:
I would prefer a snapshot here, when we are not interested in specific properties

Reply:
Thought about this, but it would be too easy to lose sight of the important bits, such as the types etc. So changed it to ensure the actually important parts, as you're right, we're not interested in the specific values

Reply:
Looks good :+1: